doc/manual/source/store/building.md

```diff
@@ -1,101 +1,187 @@
+> A derivation is a specification for running an executable on precisely defined input to produce one or more [store objects][store object].
 
-[realised]: @docroot@/glossary.md#gloss-realise
+This page describes *building* a derivation, which is to say following the instructions in the derivation to actually run the executable.
+In some cases the derivation is self-explanatory.
+For example, the arguments specified in the derivation really are the arguments passed to the executable.
+In other cases, however, there is additional procedure true for all derivations, which is therefore *not* specified in the derivation.
+This page specifies this invariant procedure that is true for all derivations, too.
+
+The chief design consideration for the building process is *determinism*.
+Conventional operating systems are typically not designed with determinism in mind.
+But determinism is needed to make Nix's caching a transparent abstraction.
+
+> **Explanation**
+>
+> For example, no one wants to slightly modify a derivation, and then find that it no longer builds for an unrelated reason, because the original derivation *also* doesn't build anymore, but the cache hit on the original derivation was hiding this.
+> We want builds that once succeed to continue succeeding, to encourage fearless modification of old build recipes.
+> Determinism is what enables things that once worked to keep working.
+
+The life cycle of a build can be broken down into 3 parts:
+
+1. Spawn the builder process with the proper environment, including the correct process arguments, environment variables, and file system state.
+
+2. Wait for the standard output and error of the process to be closed and/or the process to exit.
+   (If the standard streams are closed but the process hasn't exited, Nix will kill the process.)
+
+   Nix also logs the standard output and error of the process, but this is just for human convenience and does not influence the behavior of the system.
+   (Builder processes have no idea what the consumer of their standard output and error does with those streams, only that they are indeed consumed so buffers do not fill up and writes to them will continue to succeed.)
+
+3. Processing the outputs after the builder has exited.
 
-- Once this is done, the derivation is *normalized*, replacing each input deriving path with its store path, which we now know from realising the input.
+   The builder process on exit should have left beyond files for each output the derivation is supposed to produce.
+   The files must be processed to turn them into bona fide store objects.
+   If the processing suceeds, those store objects are associated with the derivation as (the results of) a successful build.
 
-## Builder Execution {#builder-execution}
+Step (3) happens externally, with just inert data since the process has exited or been killed by then.
+Step (1) however is best described not from Nix's perspective, but from the process's perspective.
```

Comment:
"Process" is also vague. The "build process" in general, or is that "process" in the OS sense?